### DISCUSSION TITLE
백엔드 코드 컨벤션 > 자동 줄바꿈 길이 변경 논의

### DISCUSSION THREAD

**Discussion by gracefulBrown** [woowacourse/prolog#350] [Ideas]
Hard wrap at 값에 따라서 자동 줄바꿈이 결정되는데
기존에 100자가 너무 작아서 줄바꿈이 발생하는 경우가 많이 발생하는데요
이 글자수를 조금 늘리면 어떨까요?

## Hard wrap at 설정 방법
![image](https://user-images.githubusercontent.com/46308949/131944507-befd333d-7844-49fa-a2af-9d11f7b677d0.png)

## Hard wrap at 100
![image](https://user-images.githubusercontent.com/46308949/131944538-227bce2f-9bd1-485a-abfe-29b786be57c0.png)

## Hard wrap at 120
![image](https://user-images.githubusercontent.com/46308949/131944593-9bb16ce7-4096-4e71-b278-d12f2e660d59.png)

**Comment by Hyeon9mak** [comment]
화끈하게 160자 이상까지 늘려도 좋을거 같습니다!!! 120자도 짧을 때가 있더라구요!!

유기적으로 멤버가 변하는 프로젝트라 메서드, 변수명이 자세할수록 좋을 텐데 
확실히 가로 길이 제한 때문에 멈칫멈칫할 때가 있었던거 같아요!! (특히나 테스트코드)

**Reply by sihyung92** [reply]
전 왕 긴게 좋어요 160자 재청합니다

**Comment by bperhaps** [comment]
ㅋㅋㅋㅋㅋ 저는 120자로 제안하겠습니다.
컨벤션이 100자인 이유가 과거 1024 * 720? 이 해상도에 맞춰서 였던것 같은데, 이제 모니터의 크기도 늘어나고 해상도도 올라가면서 길이도 점점 늘리는 추세인 것 같더라구요.(클린코드 출처)
120자인 이유는, spectacle기준 화면의 2/3를 채웠을 때 옆에 폴더구조창(?) 제외하고 120자정도가  딱 보이는 마지노선 같아서요!

정리하자면, 결국 길이제한 컨벤션이 있는 이유가, 한 줄을 한 눈에 보기 위함인데 경험상 맥북기준 120자정도면 편했다~~ 로 하겠습니다!

**Comment by seovalue** [comment]
저도 너무 짧아도 그렇고 너무 길어도 오히려 어디서부터 엔터쳐야하지라는 고민하게 되는 것 같아서 그 중 편했던 게 120자인 것 같아요!! 그래서 120자에 한 표 던집니다 ~~! 하지만 160자 극켬!! 은 아니어서 120자를 선호하되 160, 절충으로 140자도 괜춘 합니다 ㅎㅎ

**Comment by qhals321** [comment]
저는 사실 보는 거에는 잘 못알아차리는 편이라ㅠ 10자리 수만 아니라면 뭐든 괜찮습니당...
친누나가 10키로 빠지기 전까지는 '나 살 빠졌지?' 라고 물어보면 항상 빠진지 모르겠을정도로 보는 거에는 잘 못 알아채는..
100자리라면 저는 뭐든 환영입니다!!

**Comment by gracefulBrown** [comment]
## 투표 결과
120: 2 + 1
160: 2
기권: 1

> 저는 120에 한표하겠습니다!

## 결정 사항
- 120으로 하기
- 변수명은 의미를 유추할 수 있는 경우 생략을 한다.
- 아래의 상황이 오면 다시 논의하기
  - 120으로는 너무 많은 코드들이 줄바꿈이 된다고 느껴질 때
  - 화면이 넓어져서 120이상도 한 화면에 보일 때
